include/oboe/AudioStream.h

```diff
@@ -116,10 +116,11 @@ class AudioStream : public AudioStreamBase {
     * This cannot be set higher than getBufferCapacity().
     *
     * @param requestedFrames requested number of frames that can be filled without blocking
-    * @return resulting buffer size in frames or a Result::Error
+    * @return the resulting buffer size in frames (obtained using value()) or an error (obtained
+    * using error())
     */
-    virtual Result setBufferSizeInFrames(int32_t requestedFrames) {
-        return Result::ErrorUnimplemented;
+    virtual ResultWithValue<int32_t> setBufferSizeInFrames(int32_t requestedFrames) {
+        return ResultWithValue<int32_t>(Result::ErrorUnimplemented);
```

Comment:
I would prefer to be able to write just `return Result::ErrorUnimplemented` here. Maybe not use an explicit constructor for `ResultWithValue`.

Reply:
Done. `ResultWithValue` can now be constructed implicitly using a `Result`.